### PR TITLE
Feature Flags: Use SafeLocalStorage

### PR DESCRIPTION
**Pull request by danielcliu-ifixit**
## Overview

We were getting a lot of[ Sentry Errors ](https://ifixit.sentry.io/issues/4633989108/?referrer=github_integration)about accessing localstorage for the feature flags. Luckily, we've dealt with this same type of issue before, and we have SafeLocalStorage. So lets just use that!

## QA

Do feature flags still work?

CC @andyg0808 
Closes: https://github.com/iFixit/ifixit/issues/50796

### CODE REVIEW

File: packages/feature_flags/index.ts [hunk 1]
```diff
@@ -13,11 +15,11 @@ export function checkFlag(flagName: FlagKey) {
...
    const enableFlag = 'enable-' + flagName;
 
+   const safeLocalStorage = useSafeLocalStorage();
+
    if (typeof flagValue === 'object') {
-      if (flagValue.userToggle && typeof localStorage !== 'undefined') {
-         if (localStorage?.getItem(enableFlag) === 'true') {
-            return true;
-         }
+      if (safeLocalStorage?.getItem(enableFlag) === 'true') {
```
Comment (andyg0808) [inline packages/feature_flags/index.ts:21]: Are we guaranteed that `useSafeLocalStorage` returns not-`undefined` even on browsers without `localStorage` support?

**Review by andyg0808 — APPROVED** [review]
CR :+1: 
deploy_block :+1: on question.

File: packages/feature_flags/index.ts (hunk 1 above)
Comment (danielcliu-ifixit) [reply packages/feature_flags/index.ts:21]: yes. its basically just an object that wraps calls to localstorage

**Comment by danielcliu-ifixit** [comment]
un_deploy_block :v:

**Comment by deltuh-vee** [comment]
QA 🎬 
Feature flags are still working when edited locally.

**Pull request merged**